docs/source/python/api/compute.rst

```diff
@@ -45,6 +45,21 @@ Aggregations
    tdigest
    variance
 
+Cumulative Functions
+--------------------
+
+Cumulative functions are vector functions that perform a running total on its
+input and outputs an array containing the corresponding intermediate running values.
```

Comment:
I am worried that users are going to misunderstand what this function does. Is there some way we can clarify that it does _not_ iterate down a column accumulating the values in the rows of the column? I think my earlier comment about this was misunderstood.

Reply:
I'm still a little confused about this, actually, and the example above leaves me more confused. Here's what it currently does:

```python
>>> import pyarrow.compute as pc
>>> pc.cumulative_sum([1, 2, 3, 4])
<pyarrow.lib.Int64Array object at 0x7f586b71d100>
[
  1,
  3,
  6,
  10
]
```

Is this what's expected?

Reply:
>Is this what's expected
Yes I think so.

I am worried that users will will _think_ this function does this:
```python
>>> import pyarrow as pa
>>> import pyarrow.compute as pc
>>> t = pa.table({'x':[1, 2, 3, 4]})
>>> pc.cumulative_sum(t, ['x'])
pyarrow.Table
x: int64
----
x: [[1,3,6,10]]
```

Reply:
Ah, ok, so it's about the kinds of inputs expected. We can rewrite this to be explicit that it operates on and returns (chunked) arrays.

Reply:
We captured the behavior of cumsum for Pandas, R, numpy in [JIRA](https://issues.apache.org/jira/browse/ARROW-13530) and there are differences among them. Current implementation default behavior is consistent with Python and R.

Also, the only inputs supported in this PR are `Array` and `ChunkedArray`. Support for `RecordBatch` and `Table` inputs can be added in this PR or as a follow-up one. I agree that if inputs are limited, it should be stated explicit in the wordings.